go/chat/boxer.go

```diff
@@ -1036,8 +1036,42 @@ func (b *Boxer) attachMerkleRoot(ctx context.Context, msg *chat1.MessagePlaintex
 	return nil
 }
 
+func (b *Boxer) preBoxCheck(ctx context.Context, messagePlaintext chat1.MessagePlaintext) error {
+
+	typ, err := messagePlaintext.MessageBody.MessageType()
+	if err != nil {
+		return err
+	}
+	e := func(format string, args ...interface{}) error {
+		return errors.New(fmt.Sprintf("malformed %v message: ", typ) + fmt.Sprintf(format, args...))
+	}
+	if typ == chat1.MessageType_DELETEHISTORY {
```

Comment:
I usually lie using `switch-case` for checks like this, even if there is only one "real" case.